### PR TITLE
CheckPhish - Added reliability

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: demisto/etc#29347

## Description
Added reliability

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by bakatzir — COMMENTED** [review]
Please add a UT with the good_disp/susp_disp/bad_disp params.
I'm thinking that the refactor, as currently implemented, might ignore them.

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish.py [hunk 1]
```diff
@@ -196,37 +194,67 @@ def get_result_checkphish(jobID):
...
+        reliability = DBotScoreReliability.get_dbot_score_reliability_from_str(reliability)
+    else:
+        raise Exception("Please provide a valid value for the Source Reliability parameter.")
+
+    args = {
+        'base_url': params['url'],
+        'api_key': params.get('token'),
+        'use_ssl': not params.get('insecure', False),
+        'reliability': reliability
+    }
```
Comment (bakatzir) [inline Packs/CheckPhish/Integrations/CheckPhish/CheckPhish.py:243]: AS we talked about in previous PRs, please separate params from args, and pass both as dicts to the commands.

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    'job_id': 'jobid1234',
+    'url': 'http://test.com/',
+    'brand': 'unknown',
+    'url_sha256': '33f57dd894c986dadb1f0c9bde11bea035833560d889607bf63971093abf0695',
+    'disposition': 'clean',
+    'error': False
+}
+
+
+def test_reliability_in_get_result_checkphish(requests_mock, mocker):
```
Comment (bakatzir) [inline Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:16]: please add given/when/then

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    requests_mock.post('https://developers.checkphish.ai/api/neo/scan/status', json=RESULTS)
+
+    mocker.patch.object(demisto, 'results')
+    CheckPhish.unite_dispositions('adult', 'cryptojacking', 'drug_spam')
+    CheckPhish.get_result_checkphish('jobid1234', 'apikey', 'https://developers.checkphish.ai/api/neo/scan', False,
+                                     DBotScoreReliability.B)
+
+    assert demisto.results.call_args_list[0][0][0]['Contents'] == RESULTS
+    assert demisto.results.call_args_list[0][0][0]['EntryContext']['DBotScore']['Reliability'] == DBotScoreReliability.B
```
Comment (bakatzir) [inline Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:26]: this UT is great. need to add one more to show that if `clean` disposition is added to the `bad_disp`, than the reputation is `Bad`. Let me know if i'm missing anything.

**Review by bakatzir — CHANGES_REQUESTED** [review]

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py (hunk 2 above)
Comment (yaakovpraisler) [reply Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:16]: done.

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py (hunk 3 above)
Comment (yaakovpraisler) [reply Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:47]: done.

**Review by bakatzir — CHANGES_REQUESTED** [review]
please note and tend the build warning:
`Packs/CheckPhish/Integrations/CheckPhish/CheckPhish.yml: [DB101] - The DBotScore description of the reputation command url aren't valid. Missing: {'DBotScore.Type', 'DBotScore.Score', 'DBotScore.Indicator', 'DBotScore.Vendor'}. Fix according to context standard https://xsoar.pan.dev/docs/integrations/context-standards`

Also, please split the UT to 2 separate cases and test the outputs you're adding as well.

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py [hunk 4]
```diff
@@ -12,31 +11,19 @@
...
+            - The user reliability param and the bad_disp param that exist in the default
         When:
             - Running get_result_checkphish
         Then:
             - Verify reliability as excepted
             - Verify dbot score is expected according the bad disposition
     """
     mocked_results = RESULTS
-    mocked_results['disposition'] = bad_disp
+    mocked_results['disposition'] = 'phish'
```
Comment (bakatzir) [inline Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:26]: why a `phish` disposition is a dbotsscore 3?, you would have to explian in the docstring

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py [hunk 5]
```diff
@@ -45,4 +32,26 @@ def test_reliability_in_get_result_checkphish(requests_mock, mocker, bad_disp, e
...
+        Given:
+            - The user bad_disp param which does not exist in the default
+        When:
+            - Running get_result_checkphish
+        Then:
+            - Verify reliability as excepted
+            - Verify dbot score is expected according the bad disposition
+    """
+    mocked_results = RESULTS
+    mocked_results['disposition'] = 'bad'
```
Comment (bakatzir) [inline Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:49]: why a `bad` disposition is a dbotsscore 0?, you would have to explain in the docstring.
I would expect that a bad one would result in a dbotscore of 3.
something is not clear here.

**Review by bakatzir — COMMENTED** [review]
Good work on the outputs - please look at the UT readability and purpose comments.

File: Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py (hunk 4 above)
Comment (yaakovpraisler) [reply Packs/CheckPhish/Integrations/CheckPhish/CheckPhish_test.py:26]: I have written that the given bad_disp is in the default bad_disp dict, and the second test got a bad_disp which is not in the default

**Review by bakatzir — COMMENTED** [review]
please re generate the readme to the markdown format using the SDK
`Packs/CheckPhish/Integrations/CheckPhish/README.md`

**Review by bakatzir — APPROVED** [review]

**Pull request merged**